xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-ui/src/main/resources/XWiki/SearchSuggestSourceSheet.xml

```diff
@@ -54,10 +54,10 @@
   #set ($displayFormType = $property.getProperty('displayFormType'))
   #set ($isCheckbox = $editing &amp;&amp; $property.classType == 'Boolean' &amp;&amp; (!$displayFormType || $displayFormType.value == 'checkbox'))
   ; {{html wiki="true"}}
-    &lt;label#if ($editing) for="${prefix}$property.name"#end&gt;
-      #if ($isCheckbox)
-        $doc.display($property.name, 'edit')
-      #end
+    #if ($isCheckbox)
+      $doc.display($property.name, 'edit', $object)
+    #end
+    &lt;label#if ($editing &amp;&amp; $object!={}) for="${prefix}$property.name"#end&gt;
```

Comment:
It's not very clear to me what `$object!={}` exactly does (feel like a comparison with an empty map, but `$object` is not a map, so I guess I'm missing something.).

Reply:
@tmortagne It's not easy to understand what happens here in fact :)

The issue is that this content is called in two ways:
* multiple times to generate the search suggest source panel content (regular - visible use)
* once without a proper $object on the seach suggest page load (border - not visible use).
This secondary use stores the template in a hidden DOM element that can be used further down the line to generate the DOM for new sources without reloading the page.
This content is rendered by https://github.com/xwiki/xwiki-platform/blob/742cd4591642be4cdcaf68325f17540e0934e64e/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-ui/src/main/resources/XWiki/SearchSuggestConfigSheet.xml#L89 . The `$object` takes its value from the `$source` parameter. The value of this `$source` parameter is, in most cases, [a regular object](https://github.com/xwiki/xwiki-platform/blob/742cd4591642be4cdcaf68325f17540e0934e64e/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-ui/src/main/resources/XWiki/SearchSuggestConfigSheet.xml#L160). However, for the template, which has no object, an empty map is given as a parameter instead: https://github.com/xwiki/xwiki-platform/blob/742cd4591642be4cdcaf68325f17540e0934e64e/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-ui/src/main/resources/XWiki/SearchSuggestConfigSheet.xml#L168 . If I understand well, we'd expect an empty object instead. I didn't want to create too much changes in this content, so I just rolled with it and added a condition that works with the empty map.

This condition on the object makes sure we don't add a `for` attribute to labels that would have no input bound to them.
In a few words, `$object` is not hard typed to an object, it's also sometimes a map, henceforth this weird looking condition.

Those changes are contained in the commit https://github.com/xwiki/xwiki-platform/pull/2816/commits/f73bcc7fd4d00f1a0138bd58688d4a704d3d08d2 .

Reply:
Added  a comment in https://github.com/xwiki/xwiki-platform/pull/2816/commits/8031c992b289871e936d70de3542ab02d29130d7 :+1: